File: IntroToAlgorithms/Chap2/1-InsertionSort/1-BasicISA/BasicISA.php
<?php

// Procedure to sort the array through the `Insertion Sort` algorithm
function insertionSort(&$arr){// we need to pass the array by reference to affect its order
    for($i = 0; $i < count($arr); $i++){
    $key = $arr[$i];
    $j = $i-1;
    while($j >= 0 and $arr[$j] > $key){
        $arr[$j + 1] = $arr[$j];
        $j = $j - 1;
    }
    $arr[$j + 1] = $key;
    }
}

// Procedure to show the array elements
function showArray($arr){
    for($k = 0; $k<count($arr); $k++){
    echo $arr[$k] . " ";
    }
    echo "\n";
}

$A = array(3,7,2,8);

echo "The not-sorted array is: \n";
    
// show the array before being sorted by calling call the `showArray` procedure
showArray($A);
    
// call the `insertionSort` algorithm procedure
insertionSort($A);
    
echo "The sorted array is: \n";
// show the sorted array
showArray($A);

?>

File: IntroToAlgorithms/Chap2/1-InsertionSort/1-BasicISA/BasicISA.cpp
#include <iostream>
using namespace std;

// Procedure to sort the array through the `Insertion Sort` algorithm
void insertionSort(int arr[], int n){
	for(int i = 0; i < n; i++){
		int key = arr[i];
        int j = i - 1;
        
        while(j >= 0 && arr[j] > key){
            arr[j+1] = arr[j];
            j = j - 1;
        }
        
        arr[j+1] = key;
    }
}

// Procedure to show the array elements
void showArray(int arr[], int n){
	for(int i = 0; i < n; i++){
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main(){
    
	const int N = 4;
    int A[N] = {3,7,2,8};
    int key;
    
    cout << "The not-sorted array is: " << endl;
    
    // show the array before being sorted by calling call the `showArray` procedure
    showArray(A,N);
    
    // call the `insertionSort` algorithm procedure
    insertionSort(A,N);
    
    cout << "The sorted array is: " << endl;
    // show the sorted array
    showArray(A,N);
    
    return 0;
}
